vector/src/main/java/im/vector/app/features/home/room/list/home/HomeRoomListViewModel.kt

```diff
@@ -168,26 +202,32 @@ class HomeRoomListViewModel @AssistedInject constructor(
                 .map { it.isNotEmpty() }
                 .distinctUntilChanged()
 
-        favouritesFlow.combine(dmsFLow) { hasFavourite, hasDm ->
-            hasFavourite to hasDm
-        }.onEach { (hasFavourite, hasDm) ->
-            val filtersData = mutableListOf(
-                    HomeRoomFilter.ALL,
-                    HomeRoomFilter.UNREADS
-            )
-            if (hasFavourite) {
-                filtersData.add(
-                        HomeRoomFilter.FAVOURITES
-                )
-            }
-            if (hasDm) {
-                filtersData.add(
-                        HomeRoomFilter.PEOPlE
-                )
-            }
-
-            flow.emit(filtersData)
-        }.launchIn(viewModelScope)
+        favouritesFlow
+                .combine(dmsFLow) { hasFavourite, hasDm ->
+                    hasFavourite to hasDm
+                }.combine(filtersPreferencesFlow) { (hasFavourite, hasDm), areFiltersEnabled ->
+                    Triple(hasFavourite, hasDm, areFiltersEnabled)
```

Comment:
You can use `combine` without any prefix to combine three flows in one go
https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/combine.html

```suggestion
        combine(favouritesFlow, dmsFlow, filteresPreferencesFlow) { hasFavourite, hasDm, areFiltersEnabled ->
            Triple(hasFavourite, hasDm, areFiltersEnabled)
```

Also note capitalisation on dmsFLow